core/wasmWasi/src/internal/TimeZonesInitializer.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2019-2023 JetBrains s.r.o. and contributors.
+ * Use of this source code is governed by the Apache 2.0 License that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.datetime.internal
+
+@RequiresOptIn
+internal annotation class InternalDateTimeApi
+
+/*
+This is internal API which is not intended to use on user-side.
+ */
+@InternalDateTimeApi
+public interface TimeZonesProvider {
```

Comment:
If you mark this as `internal` and `@Suppress` the warning, will anything break? It would be nicer not to expose unnecessary symbols to autocompletion.

Reply:
Well, I would prefer not suppressing anything in libraries. Only with special request from libraries team.

Reply:
We prioritize user experience over how clean our internal implementation is, so yes, if it's possible to avoid extra public entry points in a robust manner, we try to do so.

Reply:
> @Suppress the warning

Note that accessing internals outside of the module is an error, not a warning.

Reply:
Ok, in that case, why not `@Suppress` the error?

Reply:
This is not advisable by the compiler team and can stop working in K2